deckhouse-controller/pkg/helpers/change_registry/change_registry.go

```diff
@@ -113,10 +113,17 @@ func ChangeRegistry(newRegistry, username, password, caFile, newDeckhouseImageTa
 		secretYaml, _ := yaml.Marshal(deckhouseSecret)
 		deploymentYaml, _ := yaml.Marshal(deckhouseDeploy)
 		logEntry.Println("------------------------------")
+		logEntry.Println("Deployment bashible-apiserver will be scaled to 0 replicas")
+		logEntry.Println("------------------------------")
 		logEntry.Printf("New Secret will be applied:\n%s\n", secretYaml)
 		logEntry.Println("------------------------------")
 		logEntry.Printf("New Deployment will be applied:\n%s\n", deploymentYaml)
 	} else {
+		logEntry.Println("Scaling-down bashible-apiserver...")
+		if err := scaleBashibleAPIServerDeployment(ctx, kubeCl, 0); err != nil {
```

Comment:
Looks weird. I really don't like this solution.
What for do we scale the deployment? We have a lock during the rebuilt phase, maybe we just should add some sleep there. Or fix the basic problem with reloading?
Scaling down and up is really a crutch. Also I don't see back-scaling, I think it should be done by Deckhouse

Reply:
We have discussed it with @RomanenkoDenys.
Let's create an issue, to replace this part of code with `Deckhouse Pod readiness` subscription in the bashibe-apiserver. But we could do it later, it's not urgent.